libfwupdplugin/fu-udev-device.c

```diff
@@ -1411,7 +1411,13 @@ fu_udev_device_ioctl(FuUdevDevice *self, gulong request, guint8 *buf, gint *rc,
 		return FALSE;
 	}
 
+#ifdef HAVE_ERRNO_H
+	while (((rc_tmp = ioctl(priv->fd, request, buf)) < 0) &&
+	       (errno == EINTR || errno == EAGAIN))
+		;
+#else
 	rc_tmp = ioctl(priv->fd, request, buf);
+#endif
```

Comment:
yup, that works. I'm wondering whether we also need a fu_udev_device_ioctl_full that controls this behaviour? And if so, weather we want to limit the number of retries, or have some kind of ms timeout to avoid waiting here forever for a faulty device.

Reply:
Yes I was wondering about the eventual timeout thing too. Good idea.